File: packages/revocation/python/caracalai_revocation/iface.py
# Copyright (C) 2026 Garudex Labs.  All Rights Reserved.
# Caracal, a product of Garudex Labs
#
# Revocation store contract for resource servers consulting caracal.sessions.revoke.

from __future__ import annotations

from typing import Protocol


class RevocationStore(Protocol):
    def is_revoked(self, sid: str) -> bool: ...
    def mark_revoked(self, sid: str, ttl_ms: int | None = None) -> None: ...

File: packages/revocation/go/iface.go
// Copyright (C) 2026 Garudex Labs.  All Rights Reserved.
// Caracal, a product of Garudex Labs
//
// Revocation store contract for resource servers consulting caracal.sessions.revoke.

package revocation

import (
	"context"
	"time"
)

// Store reports whether a session id has been revoked and accepts new revocations.
type Store interface {
	IsRevoked(ctx context.Context, sid string) bool
	MarkRevoked(sid string, ttl time.Duration)
}
